src/models/BaseModel.py

```diff
@@ -0,0 +1,70 @@
+import math
+
+import abc
+import pandas as pd
+import sklearn
+
+from cleaning_functions import pandas_vector_to_list
+
+
+class BaseModel:
+    def __init__(self, train_raw_data_csv_file_name='../data/Train_rev1.csv',
+                 train_normalized_location_file_name='../data/train_normalised_location.csv',
+                 train_test_ratio=0.8, load_location=True):
+        # TODO: make this static with single load
+        self.raw_data = pd.read_csv(train_raw_data_csv_file_name)
+
+        # split original training dataset into train and test data
+        # since we most likely wont get the test salary
+        self.full_dataset_size = len(self.raw_data)
+        self.training_dataset_size = math.ceil(self.full_dataset_size * train_test_ratio)
+        self.test_dataset_size = self.full_dataset_size - self.training_dataset_size
+        self.train_raw_data = self.raw_data[:self.training_dataset_size]
+        self.test_raw_data = self.raw_data[-self.test_dataset_size:]
+
+        self.train_description_feature = self.train_raw_data[['FullDescription']]
```

Comment:
can we just use two dataframes (i.e. `self.data_raw` and `self.data`)? Much cleaner.